content/modeling/modeling_2.md

```diff
@@ -0,0 +1,108 @@
+
+***
+
+#### **Step 1:** Set-up Variables
+
+The fractribution dbt package comes with a list of variables specified with a default value that you may need to overwrite in your own dbt project `dbt_project.yml` file. For details you can have a look at the installed package's default variables which can be found at `[dbt_project_name]/dbt_packages/fractribution/dbt_project.yml`.
+
+For the sake of simplicity we have selected the variables that you will most likely need to overwrite, the rest can be changed at a later stage if and when it is needed.
+
+- `conversion_window_start_date`: The start date in UTC for the window of conversions to include
+- `conversion_window_end_date`: The end date in UTC for the window of conversions to include
+- `conversion_hosts`: url_hosts to consider
+- `path_lookback_steps`: The limit for the number of marketing channels to look at before the conversion (default is 0 = unlimited)
+- `path_lookback_days`: Restrict the model to marketing channels within this many days of the conversion (values of 30, 14 or 7 are recommended)
+- `path_transforms`: An array of path transforms and their arguments (see below section **Path Transform Options**)
+- `consider_intrasession_channels`: Boolean. If false, only considers the channel at the start of the session (i.e. first page view). If true, considers multiple channels in the conversion session as well as historically.
+
+
+Below is an example snippet of these variables in `dbt_project.yml`:
+
+```yml
+vars:
+  conversion_window_start_date: '2022-07-03'
+  conversion_window_end_date: '2022-08-01'
+  conversion_hosts: ['exampleurl.com']
+  path_lookback_steps: 0
+  path_lookback_days: 30
+  path_transforms: [['Exposure', null]]
+  consider_intrasession_channels: false 
+```
+
+**Path Transform Options**
+
+Paths to conversion are often similar, but not identical. As such, path transforms reduce unnecessary complexity in similar paths before running the attribution algorithm.
+
+There are five path transform options available:
+- Exposure (default in this package): the same events in succession are reduced to one: A -> A -> B becomes A -> B. A compromise between first and unique.
+- Unique: all events in a path are treated as unique (no reduction of complexity). Best for smaller datasets (small lookback window) without a lot of retargeting.
+- First: keep only the first occurrence of any event: A -> B -> A becomes A -> B. Best for brand awareness marketing.
+- Frequency: keep a count of the events' frequency: A -> A -> B becomes A(2) -> B. Best when there is a lot of retargeting.
+- Recency: time bucket events depending on when they occurred before the conversion and collapse those in the same bucket (square braces indicate minutes before conversion event): A[15-30] -> B[1-5] -> A[1-5] -> B[1-5] becomes A[15-30] -> B[1-5] -> A[1-5]. Best for larger lookback windows.
+
+#### **Step 2:** Configure macros
```

Comment:
I have noticed that the `create_snowflake_udfs() `macro does not get called, therefore when I first tried to run the model it failed due to TRIMLONGPATH, EXPOSURE not existing in our Snowflake db yet. I have added it to the beginning of the first model that gets run (s_conversions_by_customer_id) for it to run locally for now.

Reply:
Ah yes - in the accelerator we say to run this udf, but you're right, it should be run as part of the package being run - otherwise anyone using the package without following the accelerator docs will get the same error. I have now called this macro with the on-run-start hook. Thanks!